kotlinx-coroutines-core/jvm/src/Future.kt

```diff
@@ -5,42 +5,22 @@ package kotlinx.coroutines
 
 import java.util.concurrent.*
 
-/**
- * Cancels a specified [future] when this job is cancelled.
- * This is a shortcut for the following code with slightly more efficient implementation (one fewer object created).
- * ```
- * invokeOnCompletion { if (it != null) future.cancel(false) }
- * ```
- *
- * @suppress **This an internal API and should not be used from general code.**
- */
-@InternalCoroutinesApi
-public fun Job.cancelFutureOnCompletion(future: Future<*>): DisposableHandle =
-    invokeOnCompletion(handler = CancelFutureOnCompletion(future))
-
 /**
  * Cancels a specified [future] when this job is cancelled.
  * This is a shortcut for the following code with slightly more efficient implementation (one fewer object created).
  * ```
  * invokeOnCancellation { if (it != null) future.cancel(false) }
  * ```
  */
+@Deprecated(
+    "This function does not do what its name implies: it will not cancel the future if just cancel() was called.",
```

Comment:
> it will not cancel the future if just cancel() was called

Can you please specify the difference between the case when the method does work and does not work? For example, `The future will be cancelled only when the job is cancelled, but not the Continuation`